kv/kvserver: TestGossipHandlesReplacedNode failed

kv/kvserver.TestGossipHandlesReplacedNode [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16378965?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16378965?buildTab=artifacts#/) on release-24.2 @ [e102b02a24a6a08d63d34901d073fcab075106c2](https://github.com/cockroachdb/cockroach/commits/e102b02a24a6a08d63d34901d073fcab075106c2):


```
=== RUN   TestGossipHandlesReplacedNode
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/07c61ad38d09a102cdbc84242fae185d/logTestGossipHandlesReplacedNode1837888993
    test_log_scope.go:81: use -show-logs to present logs inline
    testcluster.go:687: timed out waiting for 4 store descriptors: map[1:{} 2:{} 3:{}]
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/07c61ad38d09a102cdbc84242fae185d/logTestGossipHandlesReplacedNode1837888993
--- FAIL: TestGossipHandlesReplacedNode (49.35s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #127576 kv/kvserver: TestGossipHandlesReplacedNode failed [A-testing C-bug C-test-failure O-robot P-3 T-kv branch-release-24.1.3-rc]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGossipHandlesReplacedNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41097

--- a/pkg/kv/kvserver/gossip_test.go
+++ b/pkg/kv/kvserver/gossip_test.go
@@ -28,6 +28,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
+	"github.com/stretchr/testify/require"
 )
 
 func TestGossipFirstRange(t *testing.T) {
@@ -188,8 +189,11 @@ func TestGossipHandlesReplacedNode(t *testing.T) {
 	// beginning of the test, and to make sure they aren't closed on server
 	// shutdown. Then we can pass the listeners to the second invocation. Alas,
 	// this requires some refactoring that remains out of scope for now.
-	if err := tc.AddAndStartServerE(newServerArgs); err != nil && !testutils.IsError(err, `address already in use`) {
-		t.Fatal(err)
+	err := tc.AddAndStartServerE(newServerArgs)
+	if testutils.IsError(err, `address already in use`) {
+		skip.WithIssue(t, 114036, "could not start server due to port reuse:", err)
+	} else {
+		require.NoError(t, err)
 	}
 
 	tc.WaitForNStores(t, tc.NumServers(), tc.Server(1).GossipI().(*gossip.Gossip))

